packages/superset-ui-core/test/query/processExtraFormData.test.ts

```diff
@@ -0,0 +1,183 @@
+  overrideExtraFormData,
+} from '@superset-ui/core/src/query/processExtraFormData';
+
+describe('appendExtraFormData', () => {
+  it('should add allowed values to non-existent value', () => {
+    expect(
+      appendExtraFormData(
+        {
+          datasource: 'table_1',
+          granularity: 'something',
+          viz_type: 'custom',
+        },
+        {
+          filters: [{ col: 'my_col', op: '==', val: 'my value' }],
+        },
+      ),
+    ).toEqual({
+      datasource: 'table_1',
+      filters: [{ col: 'my_col', op: '==', val: 'my value' }],
+      granularity: 'something',
+      viz_type: 'custom',
+    });
+  });
+
+  it('should add allowed values to pre-existent value', () => {
+    expect(
+      appendExtraFormData(
+        {
+          granularity: 'something',
+          viz_type: 'custom',
+          datasource: 'table_1',
+          filters: [{ col: 'my_col', op: '==', val: 'my value' }],
+        },
+        {
+          filters: [{ col: 'my_other_col', op: '!=', val: 'my other value' }],
+        },
+      ),
+    ).toEqual({
+      granularity: 'something',
+      viz_type: 'custom',
+      datasource: 'table_1',
+      filters: [
+        { col: 'my_col', op: '==', val: 'my value' },
+        { col: 'my_other_col', op: '!=', val: 'my other value' },
+      ],
+    });
+  });
+
+  it('should add new freeform where', () => {
+    expect(
+      appendExtraFormData(
+        {
+          datasource: 'table_1',
+          granularity: 'something',
+          viz_type: 'custom',
+        },
+        {
+          extras: {
+            where: '1 = 0',
+          },
+        },
+      ),
+    ).toEqual({
+      datasource: 'table_1',
+      granularity: 'something',
+      viz_type: 'custom',
+      extras: {
+        where: '(1 = 0)',
+      },
+    });
+  });
+
+  it('should add new freeform where to existing where clause', () => {
+    expect(
+      appendExtraFormData(
+        {
+          datasource: 'table_1',
+          granularity: 'something',
+          viz_type: 'custom',
+          extras: {
+            where: 'abc = 1',
+          },
+        },
+        {
+          extras: {
+            where: '1 = 0',
+          },
+        },
+      ),
+    ).toEqual({
+      datasource: 'table_1',
+      granularity: 'something',
+      viz_type: 'custom',
+      extras: {
+        where: '(abc = 1) AND (1 = 0)',
+      },
+    });
+  });
+});
+
+describe('overrideExtraFormData', () => {
+  it('should assign allowed non-existent value', () => {
+    expect(
+      overrideExtraFormData(
+        {
+          granularity: 'something',
+          viz_type: 'custom',
+          datasource: 'table_1',
+        },
+        {
+          time_grain_sqla: 'PT1H',
+        },
+      ),
+    ).toEqual({
+      granularity: 'something',
+      viz_type: 'custom',
+      datasource: 'table_1',
+      time_grain_sqla: 'PT1H',
+    });
+  });
+
+  it('should override allowed pre-existing value', () => {
```

Comment:
```suggestion
  it('should override allowed preexisting value', () => {
```
... I'll just show myself out now.

Reply:
OK, so it depends on whose rulebook you want to follow: https://madelynsmark.wordpress.com/2019/01/02/pre-existing-or-preexisting-ap-and-chicago-style/

This is a silly language.